### PR TITLE
Feat/2.5 pause one reserve

### PR DESCRIPTION
**Pull request by dhadrien**
#97

### CODE REVIEW

**Review by alisscco — APPROVED** [review]

**Review by alisscco — APPROVED** [review]

File: contracts/protocol/lendingpool/LendingPool.sol [hunk 1]
```diff
@@ -720,6 +720,8 @@ contract LendingPool is VersionedInitializable, ILendingPool, LendingPoolStorage
   ) external override whenNotPaused {
     require(msg.sender == _reserves[asset].aTokenAddress, Errors.LP_CALLER_MUST_BE_AN_ATOKEN);
```
Comment (The-3D) [inline contracts/protocol/lendingpool/LendingPool.sol:721]: you can move this require in validateTransfer()

File: contracts/protocol/lendingpool/LendingPoolConfigurator.sol [hunk 2]
```diff
@@ -414,6 +425,34 @@ contract LendingPoolConfigurator is VersionedInitializable, ILendingPoolConfigur
     emit ReserveUnfrozen(asset);
   }
 
+  /**
+   * @dev Pauses a reserve. A paused reserve allow now user moves such as deposit, borrow, repay, swap interestrate, liquidate
+   * @param asset The address of the underlying asset of the reserve
+   **/
+  function pauseReserve(address asset) external onlyEmergencyOrPoolAdmin {
+    DataTypes.ReserveConfigurationMap memory currentConfig = pool.getConfiguration(asset);
```
Comment (The-3D) [inline contracts/protocol/lendingpool/LendingPoolConfigurator.sol:433]: call it reserveConfig

File: contracts/protocol/lendingpool/LendingPoolConfigurator.sol [hunk 3]
```diff
@@ -414,6 +425,34 @@ contract LendingPoolConfigurator is VersionedInitializable, ILendingPoolConfigur
...
+
+    emit ReservePaused(asset);
+  }
+
+  /**
+   * @dev Unpauses a reserve
+   * @param asset The address of the underlying asset of the reserve
+   **/
+  function unpauseReserve(address asset) external onlyEmergencyOrPoolAdmin {
+    DataTypes.ReserveConfigurationMap memory currentConfig = pool.getConfiguration(asset);
```
Comment (The-3D) [inline contracts/protocol/lendingpool/LendingPoolConfigurator.sol:447]: call it reserveConfig

File: test-suites/test-aave/reserve-pause.spec.ts [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+    VL_RESERVE_PAUSED,
+    INVALID_FROM_BALANCE_AFTER_TRANSFER,
+    INVALID_TO_BALANCE_AFTER_TRANSFER,
+  } = ProtocolErrors;
+
+  before(async () => {
+    _mockFlashLoanReceiver = await getMockFlashLoanReceiver();
+  });
+
+  it('User 0 deposits 1000 DAI. Configurator pauses pool. Transfers to user 1 reverts. Configurator unpauses the network and next transfer succees', async () => {
```
Comment (The-3D) [inline test-suites/test-aave/reserve-pause.spec.ts:25]: grammar error, "succeeds"

**Pull request merged**